Library/Homebrew/formula_specialties.rb

```diff
@@ -18,12 +18,20 @@ def self.url(val)
 # This formula serves as the base class for several very similar
 # formulae for Amazon Web Services related tools.
 class AmazonWebServicesFormula < Formula
+
+  depends_on :java
```

Comment:
I think it's fine as-is; do any of the AWS formulae depend on different versions of Java? Also, @jacknagel may have thoughts on this as I believe he was considering killing Formula subclasses.